library/src/engine/engine.ts

```diff
@@ -83,117 +159,120 @@ export class Engine {
-          apply: this.apply.bind(this),
-          cleanup: this.#cleanup.bind(this),
-          actions: this.#actions,
-          genRX: () => this.#genRX(ctx, ...(plugin.argNames || [])),
-          plugin,
-          el,
-          rawKey,
-          key,
-          value,
-          mods: new Map(),
-        }
+        this.#applyAttributePlugin(el, rawKey)
+      }
+    })
+  }
 
-        // Check the requirements
-        const keyReq = plugin.keyReq || Requirement.Allowed
-        if (hasKey) {
-          if (keyReq === Requirement.Denied) {
-            throw runtimeErr(`${plugin.name}KeyNotAllowed`, ctx)
-          }
-        } else if (keyReq === Requirement.Must) {
-          throw runtimeErr(`${plugin.name}KeyRequired`, ctx)
-        }
-        const valReq = plugin.valReq || Requirement.Allowed
-        if (hasValue) {
-          if (valReq === Requirement.Denied) {
-            throw runtimeErr(`${plugin.name}ValueNotAllowed`, ctx)
-          }
-        } else if (valReq === Requirement.Must) {
-          throw runtimeErr(`${plugin.name}ValueRequired`, ctx)
-        }
+  #applyAttributePlugin(el: HTMLorSVGElement, rawKey: string) {
+    // Find the plugin that matches, since the plugins are sorted by length descending and alphabetically
+    // the first match will be the most specific
+    const plugin = this.#plugins.find((p) => rawKey.startsWith(p.name))
 
-        // Check for exclusive requirements
-        if (
-          keyReq === Requirement.Exclusive ||
-          valReq === Requirement.Exclusive
-        ) {
-          if (hasKey && hasValue) {
-            throw runtimeErr(`${plugin.name}KeyAndValueProvided`, ctx)
-          }
-          if (!hasKey && !hasValue) {
-            throw runtimeErr(`${plugin.name}KeyOrValueRequired`, ctx)
-          }
-        }
+    // Skip if no plugin is found
+    if (!plugin) return
 
-        for (const rawMod of rawModifiers) {
-          const [label, ...mod] = rawMod.split('.')
-          ctx.mods.set(
-            camelize(label),
-            new Set(mod.map((t) => t.toLowerCase())),
-          )
-        }
+    // Ensure the element has an id
+    if (!el.id.length) el.id = elUniqId(el)
 
-        // Load the plugin and store any cleanup functions
-        const removal = plugin.onLoad(ctx)
-        if (removal) {
-          if (!this.#removals.has(el)) {
-            this.#removals.set(el, {
-              id: el.id,
-              fns: [],
-            })
-          }
-          this.#removals.get(el)?.fns.push(removal)
-        }
+    // Extract the key and value from the dataset
+    let [key, ...rawModifiers] = rawKey.slice(plugin.name.length).split(/\_\_+/)
+
+    const hasKey = key.length > 0
+    if (hasKey) {
+      // Keys starting with a dash are not converted to camel case in the dataset
+      const keySlice1 = key.slice(1)
+      key = key.startsWith('-') ? keySlice1 : key[0].toLowerCase() + keySlice1
+    }
+    const value = `${el.dataset[rawKey]}` || ''
```

Comment:
If `el.dataset[rawKey]` is undefined, this would result in a value of `"undefined"` and not an empty string. Is that intended?

Perhaps this is desired instead:
```suggestion
    const value = el.dataset[rawKey] || ''
```